components/datetime/src/neo_marker.rs

```diff
@@ -1869,11 +1884,11 @@ macro_rules! impl_zone_marker {
         /// .unwrap();
         ///
         /// // Time zone for America/Chicago in the summer
-        /// let zone = CustomTimeZone::from_parts(
-        ///     -40, // offset eighths of hour
-        ///     tinystr!(8, "uschi"), // time zone ID
-        ///     tinystr!(4, "amce"), // metazone ID
-        ///     tinystr!(2, "dt"), // zone variant: daylight time
+        /// let zone = TimeZone::new("-05:00".parse().unwrap(), TimeZoneBcp47Id(tinystr!(8, "uschi")))
+        ///     .to_formattable_at(
+        ///         &Default::default(),
+        ///         &Default::default(),
+        ///         &icu_calendar::DateTime::try_new_iso_datetime(2024, 7, 1, 0, 0, 0).unwrap()
```

Comment:
Issue: this new code is less readable than the old code, which was already not very readable.

#5554